### PR TITLE
[NPU] Add NpuSRTPlatform so current_platform resolves on Ascend

### PR DESCRIPTION
**Pull request by Ray-RP**
## Motivation

The `platforms/` subsystem shipped `PlatformEnum.NPU` and the `npu -> hccl` distributed-backend mapping, but the concrete `platforms/npu.py` module and the corresponding fallback branch in `_resolve_platform` were never landed. On an Ascend 910B / A2 host with `torch_npu` installed and no `SGLANG_PLATFORM` set, `sglang.srt.platforms.current_platform` therefore fell back to the base `SRTPlatform` (device_type='cpu', backend='gloo') even though the full `hardware_backend/npu/` stack (ascend attention, NPUMHATokenToKVPool, NPUGraphRunner, ...) was present in the tree.

Related RFC: #20372 ("Introduce hardware plugin system") explicitly calls for `platforms/npu.py`.

## Modifications

This change fills that gap in-tree, matching the design intent already visible in `platforms/device_mixin.py` and mirroring `platforms/xpu.py`:

** New `python/sglang/srt/platforms/npu.py` ** (mirrors `platforms/xpu.py`): 
- `NpuDeviceMixin(DeviceMixin)` overriding the memory / device / seed helpers against `torch.npu.*`, with a lazy `_ensure_torch_npu()` guard so environments that don't have `torch_npu` installed never trigger a hard import at module load time. 
- `NpuSRTPlatform(NpuDeviceMixin, SRTPlatform)` overriding the NPU-specific policy bits:  
      - `get_default_attention_backend()` ->`"ascend"`, 
      - `get_dispatch_key_name()` -> `"npu"`,
      - `supports_fp8()` and `support_cuda_graph()` -> True,
      - `support_piecewise_cuda_graph()` -> False.
- `get_torch_distributed_backend_str` is intentionally  **not** overridden -- `_DEVICE_TO_DISTRIBUTED_BACKEND` in `device_mixin.py` already maps `"npu" -> "hccl"` (or `"zbal"` when `SGLANG_ZBAL_LOCAL_MEM_SIZE > 0`).
- The `get_*_cls` factory methods are left at their base `NotImplementedError` defaults on purpose: in-tree NPU dispatch already flows through `if _is_npu:` guards in `kv_cache_configurator`, `cuda_graph_setup`, etc., and the factories are only consulted for out-of-tree platforms via `current_platform.is_out_of_tree()`. This matches how `CudaSRTPlatform` handles the same set of methods.

** `python/sglang/srt/platforms/__init__.py` **: 
- Import `NpuSRTPlatform`. 
- Add `_is_npu_available()`, which checks `hasattr(torch, "npu") and torch.npu.is_available()`. It deliberately does NOT call `sglang.srt.utils.common.is_npu`, because that helper raises when `torch_npu` is importable but no NPU is visible, and a boolean "available" probe must never raise. 
- Wire a fallback branch in `_resolve_platform` after the XPU branch and before the base fallback, and document the new arm in the discovery-flow docstring.

## Verification

Verified on an 8x Ascend 910B2 host (torch 2.10, torch_npu 2.10.0, CANN 9.0.0). With no `SGLANG_PLATFORM` set and no plugin registered:

```python       
  >>> from sglang.srt.platforms import current_platform
  >>> type(current_platform).__name__                                                                                                                                                                          
  'NpuSRTPlatform'
  >>> current_platform.device_type, current_platform.is_npu()                                                                                                                                                  
  ('npu', True)   
  >>> current_platform.get_torch_distributed_backend_str()
  'hccl'                                                                                                                                                                                                       
  >>> current_platform.get_default_attention_backend()
  'ascend'                                                                                                                                                                                                     
  >>> current_platform.get_device_name(0)
  'Ascend910B2'
  >>> current_platform.get_device_total_memory(0)
  65452113920
```

CUDA / ROCm / XPU / CPU-opt-in fallback paths are unchanged. No existing in-tree file is modified besides platforms/__init__.py, so hardware_backend/npu/* and every is_npu() legacy call site continue to work exactly as before.                                                                                                                                                                                                                                                                          
                                                                                                                                                                                         
## Checklist       
                                                                                                                                                                                                               
- Format your code according to the https://docs.sglang.ai/developer_guide/development_guide_using_docker.html#code-formatting-with-pre-commit.                                                              
- Add unit tests as outlined in the https://docs.sglang.ai/developer_guide/development_guide_using_docker.html#running-unit-tests-adding-to-ci — N/A for a device-registration patch; covered by manual verification above on real 910B2 hardware.
- Update documentation as needed, including docstrings or example tutorials.


<!-- pr-states:start -->
---
### CI States

Latest PR Test (Base): <!-- slot:pr-test:start -->:x: [Run #31392979514](https://github.com/sgl-project/sglang/actions/runs/31392979514)<!-- slot:pr-test:end -->
Latest PR Test (Extra): <!-- slot:pr-test-extra:start -->:x: [Run #31392978004](https://github.com/sgl-project/sglang/actions/runs/31392978004)<!-- slot:pr-test-extra:end -->
<!-- pr-states:end -->

### CODE REVIEW

**Comment by litmei** [comment]
Should add ut like other platforms on `test/registered/unit/platforms/test_platform_interface.py`.

File: python/sglang/srt/platforms/npu.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    device_name: str = "npu"
+    device_type: str = "npu"
+
+    @staticmethod
+    def _ensure_torch_npu() -> None:
+        # torch_npu registers the ``npu`` device with the torch dispatcher on
+        # import. It's not a hard dependency of sglang, so we import here on
+        # demand and give a friendly error if it's missing.
+        if not hasattr(torch, "npu"):
+            import torch_npu  # noqa: F401  # pylint: disable=import-outside-toplevel
```
Comment (litmei) [inline python/sglang/srt/platforms/npu.py:35]: Can we move this check to `self.__init__` so that `import torch_npu` is called only once, instead of repeatedly calling it in multiple methods?

Comment (litmei) [inline python/sglang/srt/platforms/npu.py:35]: On the other hand, PyTorch has supported automatic backend loading since version 2.5.1. Given that SGLang currently requires PyTorch 2.13.0 (as specified in pyproject.toml), and the Ascend-provided images already ship with PyTorch 2.10.0, could we consider removing the manual import torch_npu?

File: python/sglang/srt/platforms/__init__.py [hunk 2]
```diff
@@ -46,6 +47,16 @@ def _is_xpu_available() -> bool:
...
 
 
+def _is_npu_available() -> bool:
+    # Guard on hasattr because ``torch.npu`` is registered by ``torch_npu``
+    # at import time; environments without torch_npu installed simply won't
+    # have the attribute. We intentionally do NOT call
+    # ``sglang.srt.utils.common.is_npu`` here -- that helper raises when
+    # torch_npu is present but no NPU device is visible, and a boolean
+    # "available" probe must never raise.
+    return hasattr(torch, "npu") and torch.npu.is_available()
```
Comment (litmei) [inline python/sglang/srt/platforms/__init__.py:57]: PyTorch has supported automatic backend loading since version 2.5.1. Given that SGLang currently requires PyTorch 2.13.0 (as specified in pyproject.toml), and the Ascend-provided images already ship with PyTorch 2.10.0, could we consider just using `return torch.npu.is_available()`?
